docs/src/main/asciidoc/security-built-in-authentication.adoc

```diff
@@ -122,6 +122,50 @@ in the xref:resteasy-reactive.adoc[RESTEasy Reactive] from endpoints annotated w
 or with respective configuration authorization checks as authentication has already happened. The same is also valid
 for the xref:reactive-routes.adoc[Reactive routes] if a route response is synchronous.
 
+CAUTION: xref:security-authorization.adoc#standard-security-annotations[Standard security annotations] on CDI beans are not supported on IO thread if a non-void secured method returns a value synchronously and proactive authentication is disabled, as they need to access the `SecurityIdentity`.
+In the example below, we have defined `HelloResource` and `HelloService`. It's easy to see that any GET request to `/hello` will run on IO thread and throw `BlockingOperationNotAllowedException` exception.
```

Comment:
@michalvavrik The example is clear, but what is our advice here to  have it working, move `@RolesAllowed("admin")` to `HelloResource#hello` ?

Reply:
I kinda thought they will figure it out :-D (not a rocket science), but you are right, I've a few suggestions. For formatting reasons, it's not a CAUTION anymore (I need a list).

Reply:
Hey @michalvavrik That is a nice summary of 3 available options to get it fixed, thanks